ConfigSpace/hyperparameters.pyx

```diff
@@ -1538,88 +1555,129 @@ cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
+            converted to the [0, 1] range which can be done with ``_inverse_transform``.
+
+        rs: RandomState
+            The random state to use
+
+        number: int = 4
+            How many neighbors to get
+
+        transform: bool = False
+            Whether to transform this value from the unit cube, back to the
+            hyperparameter's specified range of values.
+
+        std: float = 0.2
+            The std. dev. to use in the [0, 1] hypercube space while sampling
+            for neighbors.
+
+        Returns
+        -------
+        List[int]
+            Some ``number`` of neighbors centered around ``value``.
+        """
+        assert 0 <= value <= 1, (
+            "For get neighbors of UniformIntegerHyperparameter, the value"
+            " if assumed to be in the unit-hypercube [0, 1]. If this was not"
+            " the behaviour assumed, please raise a ticket on github."
+        )
+        # Convert python values to cython ones
+        cdef long long center = self._transform(value)
+        cdef long long lower = self.lower
+        cdef long long upper = self.upper
+        cdef unsigned int n_requested = number
+        cdef unsigned long long n_neighbors = upper - lower - 1
+        cdef long long stepsize = self.q if self.q is not None else 1
+
+        neighbors = []
+
+        cdef long long v  # A value that's possible to return
+        if n_neighbors < n_requested:
+
+            for v in range(lower, center):
+                neighbors.append(v)
+
+            for v in range(center + 1, upper):
+                neighbors.append(v)
+
+            if transform:
+                return neighbors
+            else:
+                return self._inverse_transform(np.asarray(neighbors)).tolist()
+
+
+        # A truncated normal between 0 and 1, centered on the value with a scale of std.
+        # This will be sampled from and converted to the corresponding int value
+        # OPTIM: Use of clipping insead of scipy.truncnorm
+        # Clipping is faster for generation but more likely to generate clumped values of 0/1
+        # at the boundaries, given a wide enough std and a value close to 0 or 1.
+        # However the overhead of the scipy call seems to outweight this clumping + deduping
+        # procedure in our used benchmark.
+        float_indices = np.clip(rs.normal(value, std, n_requested), 0, 1)
+
+        cdef long long [:] possible_neighbors = self._transform_vector(float_indices).astype(int)
+
+        # We make sure to find duplicates, and only try to find new neighbors for those
+        # that are not duplicates
+        cdef set seen = {center}
+        cdef set duplicates = set()
+        for i in range(n_requested):
+            v = possible_neighbors[i]
+            if v in neighbors or v == center:
+                duplicates.add(v)
+            else:
+                seen.add(v)
+                neighbors.append(v)
```

Comment:
Another slight optimization is to not enter the loop for most of the sampled values.

For a brief example, consider the antogonistic case where `possible_neighbors` was `[1, 1, 2, 3, 4, 5]` with the bounds being `(1, 6)`.

* The first `1` would have been added to `neighbors`.
* The second `1` would then enter the recovery case and become a `2`.
* The loop would move on to `2` and find it has also been seen, entering the recovery case.
* and so on...

Now we explicitly make sure to only perform this recovery option on duplicates.